### PR TITLE
Updated appveyor script to do source-indexing and nuget pack

### PR DESCRIPTION
**Pull request by ivan-danilov**

### CODE REVIEW

**Comment by JakeGinnivan** [comment]
Just realised I dont have caching in shouldly, but you can cache packages/ for faster builds:

see https://github.com/TestStack/TestStack.BDDfy/blob/master/appveyor.yml#L27

File: appveyor.yml [hunk 1]
```diff
@@ -1,12 +1,28 @@
...
+build_script:
+  - cmd: nuget restore src\TestStack.White.sln
+  - cmd: gitversion /l console /output buildserver /updateAssemblyInfo
+  
+  - cmd: msbuild src\TestStack.White.sln "/p:Configuration=%CONFIGURATION%;Platform=%PLATFORM%"
+
+  - cmd: ECHO GitLink src\ -u https://github.com/TestStack/White -c %CONFIGURATION% 
+  - cmd: GitLink . -u https://github.com/TestStack/White -c %CONFIGURATION%
+
+  - cmd: ECHO nuget pack src\TestStack.White\bin\%CONFIGURATION%\TestStack.White.nuspec -version "%GitVersion_NuGetVersion%" -prop "target=%CONFIGURATION%"
```
Comment (JakeGinnivan) [inline appveyor.yml]: Maybe better to add the source file path into the nuspec. I dont think the init and other files are copied to output?

Comment (ivan-danilov) [inline appveyor.yml]: I was going to do it, but those paths depend on configuration, and I don't know how to make NuGet replace configuration placeholder with something...

Comment (ivan-danilov) [inline appveyor.yml]: P.S. I checked - all files that are going to be packed have "Copy Always" set.

**Pull request merged**